code/modules/religion/rites.dm

```diff
@@ -218,3 +218,81 @@
 		new /obj/item/candle/infinite(altar_turf)
 	playsound(altar_turf, 'sound/magic/fireball.ogg', 50, TRUE)
 	return ..()
+
+/**** Carp rites ****/
+
+/datum/religion_rites/summon_carp
+	name = "Summon Carp"
+	desc = "Creates a Sentient Space Carp, if a soul is willing to take it."
+	ritual_length = 90 SECONDS
+	ritual_invocations = list("Carp'sie grant us a new follower ...",
+	"... let them enter our realm ...",
+	"... become one with our world ...",
+	"... to swim in our space ...",
+	"... and help our cause ...")
+	invoke_msg = "... We summon thee, Holy Carp!"
+	favor_cost = 500
+
+/datum/religion_rites/summon_carp/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	var/altar_turf = get_turf(religious_tool)
+	new /obj/effect/temp_visual/bluespace_fissure/long(altar_turf)
+	var/jobbans = list(ROLE_BRAINWASHED, ROLE_DEATHSQUAD, ROLE_DRONE, ROLE_LAVALAND, ROLE_MIND_TRANSFER, ROLE_POSIBRAIN, ROLE_SENTIENCE)
+	var/list/candidates = pollGhostCandidates("Do you wish to be summoned as a Holy Carp?", jobbans, null, FALSE,)
+	if(!candidates.len)
+		to_chat(user, "<span class='warning'>The carp pool is empty...")
+		new /obj/effect/gibspawner/generic(altar_turf)
+		user.visible_message("<span class='warning'>The carp pool was not strong enough to bring forth a space carp.")
+		GLOB.religious_sect?.adjust_favor(400, user)
+		return NOT_ENOUGH_PLAYERS
+	var/mob/dead/observer/selected = pick_n_take(candidates)
+	var/datum/mind/M = new /datum/mind(selected.key)
+	var/carp_species
+	carp_species = pick(/mob/living/simple_animal/hostile/carp/megacarp, /mob/living/simple_animal/hostile/carp)
+	var/mob/living/simple_animal/hostile/carp = new carp_species(altar_turf)
+	var/carpname = "Holy Space-Carp ([rand(1,999)])"
+	carp.name = carpname
+	carp.key = selected.key
```

Comment:
Not sure if you actually have to set this manually.

Reply:
The ckey

Reply:
ckey was the only way I managed to get the player into the mob, for whatever reason transfer_mind and a few other bits didnt work

Reply:
Should be fine, nevermind.